LeaderboardBackend/Authorization/UserTypeAuthorizationHandler.cs

```diff
@@ -35,7 +38,13 @@ UserTypeRequirement requirement
 			return Task.CompletedTask;
 		}
 
-		User? user = UserService.GetUserFromClaims(context.User).Result;
+		Guid? userId = AuthService.GetUserIdFromClaims(context.User);
+		if (userId is null)
+		{
+			context.Fail();
+			return Task.CompletedTask;
+		}
+		User? user = UserService.GetUserById(userId.Value).Result;
```

Comment:
Do we need `.Value` here? Can't we just use `userId`?

Reply:
`Guid?` is not the same as a nullable reference type. `Guid` is considered a primitive. For those, you have to get `.Value` out of it specifically.